File: Two-Pointer/713-subarray-product-less-than-k/subarray-product-less-than-k.cpp
class Solution {
public:
    int numSubarrayProductLessThanK(vector<int>& nums, int k) {
        if(k <= 1) return 0;
        int product = 1, ans = 0, i = 0, j = 0;

        while(i < nums.size()){
            product *= nums[i];

            while(product >= k){
                product = product / nums[j];
                j++;
            }

            ans += i - j + 1;
            i++;
        }

        return ans;
    }
};

File: Sliding_Window/713-subarray-product-less-than-k/subarray-product-less-than-k.py
class Solution:
    def numSubarrayProductLessThanK(self, nums: List[int], k: int) -> int:
        if k <= 1:
            return 0
        
        product = 1
        ans = 0
        j = 0

        for i in range(len(nums)):
            product *= nums[i]

            while product >= k:
                product //= nums[j]
                j += 1

            ans += i - j + 1

        return ans